pkg/controller/pod/pod_controller.go

```diff
@@ -76,8 +77,23 @@ type ReconcilePolicy struct {
 // Reconcile reads that state of the cluster for a Pod object and makes changes based on the state
 func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.V(1).Info("got a pod event in the controller")
+	instanceOptedIn, err := r.poolManager.IsPodInstanceOptedIn(request.Namespace)
```

Comment:
I would say that we check the error first so we are sure that the instanceOptedIn content is legit.